### PR TITLE
Stats annual

### PR DESCRIPTION
**Pull request by ihahalev**

### CODE REVIEW

File: database/models/transaction.model.js [hunk 1]
```diff
@@ -27,4 +27,84 @@ const transactionSchema = new mongoose.Schema(
...
+    {
+      $addFields: {
+        savings: { $subtract: ['$incomeAmount', '$expenses'] },
+        expectedSavings: {
+          $multiply: ['$incomeAmount', '$percentAmount', 0.01],
+        },
+        year: {
+          $year: {
+            date: {
+              $dateFromString: { dateString: { $concat: ['$_id', '-01'] } },
```
Comment (alucardpetrovich) [inline database/models/transaction.model.js:93]: You're receiving a complete date here (YYYY-MM-DD), not just a year

File: database/models/transaction.model.js [hunk 2]
```diff
@@ -27,4 +27,84 @@ const transactionSchema = new mongoose.Schema(
...
+            },
+          },
+        },
+        month: {
+          $month: {
+            date: {
+              $dateFromString: { dateString: { $concat: ['$_id', '-01'] } },
```
Comment (alucardpetrovich) [inline database/models/transaction.model.js:100]: You're receiving a complete date here (YYYY-MM-DD), not just a year

File: routers/transactions.controller.js [hunk 3]
```diff
@@ -19,19 +19,6 @@ class TransactionController {
 
   async createTransaction(req, res) {
     try {
-      const user = await userModel.findById(req.user);
```
Comment (alucardpetrovich) [inline routers/transactions.controller.js:22]: You need to check here at least that `req.user.familyId` is present

File: routers/transactions.controller.js [hunk 4]
```diff
@@ -68,6 +55,42 @@ class TransactionController {
...
+      );
+      return responseNormalizer(200, res, { transes });
+    } catch (e) {
+      errorHandler(req, res, e);
+    }
+  }
+
+  async familyAuthorization(req, res, next) {
+    try {
+      const user = await userModel.findById(req.user);
```
Comment (alucardpetrovich) [inline routers/transactions.controller.js]: better simply check that `req.user.familyId` is present. This check won't work right now

File: database/models/transaction.model.js (hunk 1 above)
Comment (ihahalev) [reply database/models/transaction.model.js:93]: That's the point, I'm receiving separately year and month, it was agreed with frontend

File: database/models/transaction.model.js (hunk 2 above)
Comment (ihahalev) [reply database/models/transaction.model.js:100]: That's the point, I'm receiving separately year and month, it was agreed with frontend

File: routers/transactions.controller.js (hunk 3 above)
Comment (ihahalev) [reply routers/transactions.controller.js:22]: I check it in middleware familyAuthorization before calling this method, so if there is no familyId, it woun't get here, as I understand

File: routers/transactions.controller.js (hunk 4 above)
Comment (ihahalev) [reply routers/transactions.controller.js]: const {familyId} = req.user;
if (!familyId)
Like this?

**Review by alucardpetrovich — APPROVED** [review]

**Pull request merged**